index.rst

```diff
@@ -96,6 +96,27 @@ ESPHome is a system to control your ESP8266/ESP32 by simple yet powerful configu
 
 .. _devices:
 
+General ESPHome
+---------------
+
+Automations
+***********
+
+.. raw:: html
+
+    <p>
+    <A href="/guides/automations.html">Automations</A> allow you to perform actions under certain conditions and <a href="/guides/automations.html#templates-lambdas"> templates (lambdas)</A> are a way to easily customize everything about your node, using short snippets of C++ code.
+    </p>
+
+Configuration Types
+*******************
+
+.. raw:: html
+
+    <p>
+    <a href="/guides/configuration-types.html">Configuration Types</a> covers naming of your components with "ID", how to reference GPIO pins and configure them as input or output, pullups / pull downs etc and use substitutions which are like #defines in C++.
+    </p>
+
```

Comment:
Links to these are in the `Next steps` box right above here.
I do like it and its a nice chunk of text to explain these important features, perhaps remove them from the next steps box? What do you think @glmnet @balloob ?

Reply:
Yes I know they are in Next Steps, but by adding them as headings, they appear in the TOC on the top LHS. This surfaces them (IMO) and I think lots of newbies will benefit, esp the wealth of info on the Automations page. I refer to it so much, and having it in the TOC is really handy.

Reply:
I'd say we move this part of the PR and to another? This is harder as it should work for desktop and mobile as a home page.
_I'm going back and forth on ideas here on my mind like: this!, that! then I say to myself I'm not a designer._

Reply:
I agree the automation & configuration reference could use a bit more exposition, but I'm not sure putting a block of text on the top of the homepage is the best way to do this. This is one of the first things new users see, and it doesn't sell ESPHome well (imo): with all the references to C++ and code it gives the idea that setting up a device with ESPHome is complicated, while it really isn't.

If we're just going for ease-of-access through the TOC to the pages, making a subheading "Reference" above "Cookbook" and putting them under there makes more sense imo.

Also agree with @glmnet that doing this in a separate PR is probably helpful.

Reply:
All good points, especially the bit about C++ being one of the first thing new users see.
Removed this for now. I won't pursue this is a separate PR.